### PR TITLE
Missing many error codes

### PR DESCRIPTION
**Pull request by AkiraNorthstar**
Reference: #69

### CODE REVIEW

**Comment by AkiraNorthstar** [comment]
Wow! - That was fast! ;-)

File: lib/errors.c [hunk 1]
```diff
@@ -58,25 +148,83 @@ int nterror_to_errno(uint32_t status) {
...
+        case SMB2_STATUS_LOCK_NOT_GRANTED:
+                return EDEADLK;
+        case SMB2_STATUS_OBJECT_NAME_COLLISION:
+                return EEXIST;
+        case SMB2_STATUS_PIPE_DISCONNECTED:
+                return EPIPE;
+        case SMB2_STATUS_MEDIA_WRITE_PROTECTED:
+                return EROFS;
+        case SMB2_STATUS_NO_MEDIA_IN_DEVICE:
+                return ENOMEDIUM;
```
Comment (amosavian) [inline lib/errors.c]: This error code is not avaiable in Darwin. Would you mind replace it with `ENODEV` or any other posix-defined error code?

**Comment by sahlberg** [comment]
That is fine..

Send a pull request.

**Comment by amosavian** [comment]
@sahlberg I can't send before making decision on my current open pull request. Because I forgot to branch before opening PR.

**Comment by sahlberg** [comment]
Ok, let me look at it in  two days. I am about to travel so I will be
unavailable for a couple of days.

**Pull request merged**